### ISSUE TITLE
Inconsistent results when running `cross_entropy` on CUDA and CPU

### ISSUE THREAD

**Issue by anag004** [pytorch/pytorch#70417]
### 🐛 Describe the bug

The output of `torch.nn.cross_entropy` differs when the tensors are on GPU and CPU

```
import torch
import torch.nn as nn

torch.manual_seed(seed=0)

target = torch.zeros(2, 2).long().to("cuda:1")   

target[0][0] = 3
target[1][0] = 5

logits = torch.randn(2, 10).to("cuda:1")

print(nn.functional.cross_entropy(logits.cpu(), target[:, 0].cpu(), reduction='none'))
print(nn.functional.cross_entropy(logits, target[:, 0], reduction='none'))
```

Observed Output
```
tensor([3.0468, 4.3090])
tensor([3.0468, 2.0217], device='cuda:1')
```

The two tensors should match. 

### Versions

Collecting environment information...
PyTorch version: 1.10.0+cu113
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.2 LTS (x86_64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.8.8 (default, Apr 13 2021, 19:58:26)  [GCC 7.3.0] (64-bit runtime)
Python platform: Linux-5.11.0-41-generic-x86_64-with-glibc2.10
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: 
GPU 0: NVIDIA GeForce RTX 3090
GPU 1: NVIDIA GeForce RTX 3090

Nvidia driver version: 470.57.02
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.21.4
[pip3] torch==1.10.0+cu113
[pip3] torchaudio==0.10.0+cu113
[pip3] torchvision==0.11.1+cu113
[conda] blas                      1.0                         mkl  
[conda] mkl                       2021.2.0           h06a4308_296  
[conda] mkl-service               2.3.0            py38h27cfd23_1  
[conda] mkl_fft                   1.3.0            py38h42c9631_2  
[conda] mkl_random                1.2.1            py38ha9443f7_2  
[conda] mypy_extensions           0.4.3                    py38_0  
[conda] numpy                     1.20.1           py38h93e21f0_0  
[conda] numpy-base                1.20.1           py38h7d8b39e_0  
[conda] numpydoc                  1.1.0              pyhd3eb1b0_1  
[conda] torch                     1.10.0+cu113             pypi_0    pypi
[conda] torchaudio                0.10.0+cu113             pypi_0    pypi
[conda] torchvision               0.11.1+cu113             pypi_0    pypi

**Comment by XiaobingSuper** [comment]
Hi, @anag004 , I tested your case using PyTorch nightly(```conda install pytorch torchvision torchaudio cudatoolkit=10.2 -c pytorch-nightly```), it doesn't have this isssue: 
```
tensor([3.0468, 4.3090])
tensor([3.0468, 4.3090], device='cuda:0')
```
Could you try it?

**Comment by gchanan** [comment]
I reproduced the issue on 1.10.0 and it's fixed on 1.10.1 -- please upgrade.
